whylog/tests/tests_config/test_get_search_range.py

```diff
@@ -46,33 +96,260 @@ def setUpClass(cls):
+            {
+                'clues_groups': [[1, 1], [0, 1]],
+                'name': 'time',
+                'params': {'min_delta': 10}
+            },
+        ]
+        rule = Rule([self.cause1, self.cause2], self.effect, constraints1, Rule.LINKAGE_AND)
+        calculated_ranges = self.calculate_range(rule)
+
+        expected_ranges = {
+            'database': {
+                'date': {
+                    'left_bound': self.earliest_date,
+                    'right_bound': self.ten_second_earlier
+                }
+            },
+        }
+
+        raise SkipTest('Not implemented yet')
+        assert calculated_ranges == expected_ranges
+
+    def test_search_range_lack_of_right_bound(self):
+        constraints1 = [
+            {
+                'clues_groups': [[1, 1], [0, 1]],
+                'name': 'time',
+                'params': {'max_delta': 10}
+            },
+        ]
+        rule = Rule([self.cause1, self.cause2], self.effect, constraints1, Rule.LINKAGE_AND)
+        calculated_ranges = self.calculate_range(rule)
+
+        expected_ranges = {
+            'database': {
+                'date': {
+                    'left_bound': self.ten_second_earlier,
+                    'right_bound': self.effect_time
+                }
+            },
+        }
+
+        raise SkipTest('Not implemented yet')
+        assert calculated_ranges == expected_ranges
 
-        calculated_ranges = self.config._get_search_ranges([self.rule], effect_clues)
+    def test_search_range_delayed_logs(self):
+        constraints1 = [
+            {
+                'clues_groups': [[1, 1], [0, 1]],
+                'name': 'time',
+                'params': {
+                    'min_delta': -10,
+                    'max_delta': 100
+                }
+            },
+        ]
+        rule = Rule([self.cause1, self.cause2], self.effect, constraints1, Rule.LINKAGE_AND)
+        calculated_ranges = self.calculate_range(rule)
 
         expected_ranges = {
             'database': {
                 'date': {
-                    'left_bound': datetime(2016, 4, 12, 23, 53, 3),
-                    'right_bound': datetime(2016, 4, 12, 23, 54, 33)
+                    'left_bound': self.one_hundred_second_earlier,
+                    'right_bound': self.ten_second_later
                 }
             },
+        }
+
+        raise SkipTest('Not implemented yet')
+        assert calculated_ranges == expected_ranges
+
+    def test_search_range_on_identical_constraint(self):
+        constraints1 = [{'clues_groups': [[1, 1], [0, 1]], 'name': 'identical', 'params': {}},]
+        rule = Rule([self.cause1, self.cause2], self.effect, constraints1, Rule.LINKAGE_AND)
+        calculated_ranges = self.calculate_range(rule)
+
+        expected_ranges = {
+            'database': {
+                'date': {
+                    'left_bound': self.effect_time,
+                    'right_bound': self.effect_time
```

Comment:
`left_bound` should be earliest possible time, because specified constraints does not contain any better date.